05_stringer/sahidakhtar1/main.go

```diff
@@ -0,0 +1,19 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+type IPAddr [4]byte
+
+func (i IPAddr) String() string {
+	return fmt.Sprintf("%v.%v.%v.%v", i[0], i[1], i[2], i[3])
```

Comment:
Since you know these are type `byte`, you should use `%d` here, not `%v`. May not seem like an issue but can cause issues later down the track.
If you add a type `type MyByte byte` and change `IPAddr` to `type IPAddr [4]MyByte`, the rest of the code will still work as intended. Someone can then add their own stringer method `func (m MyByte) String() string` and it will change the behaviour of this stringer method. Try out the following code
```
type MyInt int

// See the difference with and without this commented
// func (i MyInt) String() string {
// 	return "BAD"
// }

type MyMultiInt [4]MyInt

func (m MyMultiInt) String() string {
	return fmt.Sprintf("%d.%d.%v.%v", m[0], m[1], m[2], m[3])
}

func main() {
	var m = MyMultiInt{1, MyInt(1), 1, MyInt(1)}
	fmt.Println(m)
}
```

Reply:
Thanks for the nice suggestion. Changes done accordingly.